net-p2p/eiskaltdcpp/eiskaltdcpp-2.4.1.ebuild

```diff
@@ -0,0 +1,184 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-1 lua5-2 )
+
+PLOCALES="be bg cs de el en es eu fr hu it pl pt_BR ru sk sr sr@latin sv_SE tr uk vi zh_CN"
+
+inherit cmake l10n lua-single xdg-utils toolchain-funcs
+[[ ${PV} = *9999* ]] && inherit git-r3
+
+DESCRIPTION="Qt/DC++ based client for DirectConnect and ADC protocols"
+HOMEPAGE="https://github.com/eiskaltdcpp/eiskaltdcpp"
+
+LICENSE="GPL-2 GPL-3"
+SLOT="0"
+IUSE="cli daemon dbus +dht examples -gold -gtk idn -javascript +json libcanberra libnotify lua +minimal pcre +qt5 spell sqlite upnp -xmlrpc"
+
+REQUIRED_USE="
+	?? ( json xmlrpc )
+	cli? ( ^^ ( json xmlrpc ) )
+	dbus? ( qt5 )
+	javascript? ( qt5 )
+	libcanberra? ( gtk )
+	libnotify? ( gtk )
+	lua? ( ${LUA_REQUIRED_USE} )
+	spell? ( qt5 )
+	sqlite? ( qt5 )
+"
+
+if [[ ${PV} != *9999* ]]; then
+	SRC_URI="mirror://sourceforge/project/${PN}/Sources/${P}.tar.xz"
+	KEYWORDS="~amd64 ~x86"
+else
+	EGIT_REPO_URI="https://github.com/${PN}/${PN}.git"
+	KEYWORDS=""
+fi
+
+RDEPEND="
+	app-arch/bzip2
+	dev-libs/openssl:0=
+	sys-apps/attr
+	sys-libs/zlib
+	virtual/libiconv
+	virtual/libintl
+	cli? (
+		dev-lang/perl
+		dev-perl/Data-Dump
+		dev-perl/Term-ShellUI
+		virtual/perl-Getopt-Long
+		json? ( dev-perl/JSON-RPC )
+		xmlrpc? ( dev-perl/RPC-XML )
+	)
+	daemon? (
+		json? ( dev-libs/jsoncpp:= )
+		xmlrpc? ( dev-libs/xmlrpc-c[abyss,cxx] )
+	)
+	gtk? (
+		dev-libs/glib:2
+		x11-libs/gtk+:3
+		x11-libs/pango
+		x11-themes/hicolor-icon-theme
+		libcanberra? ( media-libs/libcanberra )
+		libnotify? ( x11-libs/libnotify )
+	)
+	idn? ( net-dns/libidn )
+	lua? ( ${LUA_DEPS} )
+	pcre? ( dev-libs/libpcre )
+	qt5? (
+		dev-qt/qtconcurrent:5
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtxml:5
+		dbus? ( dev-qt/qtdbus:5 )
+		javascript? (
+			dev-qt/qtdeclarative:5
+			dev-qt/qtscript:5
+		)
+		spell? ( app-text/aspell )
+		sqlite? ( dev-qt/qtsql:5[sqlite] )
+	)
+	upnp? ( net-libs/miniupnpc )
+"
+BDEPEND="
+	gold? ( sys-devel/binutils[gold] )
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+	virtual/pkgconfig
+	qt5? ( dev-qt/linguist-tools:5 )
+"
+
+DOCS=( AUTHORS ChangeLog.txt )
+
+PATCHES=(
+	"${FILESDIR}/${PN}-2.2.10-cmake_lua_version.patch"
+)
+
+CMAKE_REMOVE_MODULES_LIST="FindLua"
```

Comment:
This doesn't seem to work, 
```
[ebuild  N     ] net-p2p/eiskaltdcpp-2.4.1::testworld  USE="cli dbus dht examples json minimal pcre qt5 -daemon -gold -gtk -idn -javascript -libcanberra -libnotify -lua -spell -sqlite -upnp -xmlrpc" LUA_SINGLE_TARGET="lua5-1 -lua5-2" 4,201 KiB

Total: 1 package (1 new), Size of downloads: 4,201 KiB

Would you like to merge these packages? [Yes/No] y

>>> Verifying ebuild manifests

>>> Emerging (1 of 1) net-p2p/eiskaltdcpp-2.4.1::testworld
>>> Downloading 'https://download.sourceforge.net/project/eiskaltdcpp/Sources/eiskaltdcpp-2.4.1.tar.xz'
--2021-02-03 08:54:12--  https://download.sourceforge.net/project/eiskaltdcpp/Sources/eiskaltdcpp-2.4.1.tar.xz
Resolving download.sourceforge.net... 216.105.38.13
Connecting to download.sourceforge.net|216.105.38.13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://deac-fra.dl.sourceforge.net/project/eiskaltdcpp/Sources/eiskaltdcpp-2.4.1.tar.xz [following]
--2021-02-03 08:54:13--  https://deac-fra.dl.sourceforge.net/project/eiskaltdcpp/Sources/eiskaltdcpp-2.4.1.tar.xz
Resolving deac-fra.dl.sourceforge.net... 37.203.33.33
Connecting to deac-fra.dl.sourceforge.net|37.203.33.33|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4301580 (4.1M) [application/octet-stream]
Saving to: '/var/cache/distfiles/eiskaltdcpp-2.4.1.tar.xz.__download__'

/var/cache/distfile 100%[===================>]   4.10M   356KB/s    in 52s     

2021-02-03 08:55:16 (80.1 KB/s) - '/var/cache/distfiles/eiskaltdcpp-2.4.1.tar.xz.__download__' saved [4301580/4301580]

 * eiskaltdcpp-2.4.1.tar.xz BLAKE2B SHA512 size ;-) ...                                      [ ok ]
 * Using lua5.1 to build
>>> Unpacking source...
>>> Unpacking eiskaltdcpp-2.4.1.tar.xz to /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work
>>> Source unpacked in /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work
>>> Preparing source in /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1 ...
 * Applying eiskaltdcpp-2.2.10-cmake_lua_version.patch ...                                   [ ok ]
 * Working in BUILD_DIR: "/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build"
removed '/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1/cmake/FindLua.cmake'
 * Hardcoded definition(s) removed in CMakeLists.txt:
 *      set (CMAKE_BUILD_TYPE "RelWithDebInfo")
 * Looking in eiskaltdcpp-qt/translations for new locales ...                                [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1 ...
 * Working in BUILD_DIR: "/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build"
cmake -C /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DLIB_INSTALL_DIR=lib64 -Dlinguas=en  -DCREATE_MO=ON -DUSE_GTK=OFF -DUSE_LIBGNOME2=OFF -DUSE_QT=OFF -DUSE_QT_QML=OFF -DNO_UI_DAEMON=no -DDBUS_NOTIFY=yes -DWITH_DHT=yes -DWITH_EXAMPLES=yes -DUSE_GTK3=no -DUSE_IDNA=no -DUSE_JS=no -DUSE_LIBCANBERRA=no -DUSE_LIBNOTIFY=no -DWITH_DEV_FILES=no -DPERL_REGEX=yes -DUSE_QT5=yes -DUSE_ASPELL=no -DLOCAL_ASPELL_DATA=OFF -DUSE_QT_SQLITE=no -DUSE_MINIUPNP=no -DFORCE_XDG=ON -DENABLE_STACKTRACE=OFF -DUSE_GOLD=no -DLOCAL_JSONCPP=OFF -DBUILD_STATIC=OFF -DINSTALL_QT_TRANSLATIONS=OFF -DCOMPRESS_MANPAGES=OFF -DUSE_CLI_JSONRPC=yes -DUSE_CLI_XMLRPC=no -DWITH_EMOTICONS=ON -DWITH_SOUNDS=ON -DCMAKE_BUILD_TYPE=Gentoo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build/gentoo_toolchain.cmake  /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1
loading initial cache file /var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found PkgConfig: x86_64-pc-linux-gnu-pkg-config (found version "1.7.3") 
-- Looking for mallinfo
-- Looking for mallinfo - found
-- Looking for malloc_stats
-- Looking for malloc_stats - found
-- Looking for malloc_trim
-- Looking for malloc_trim - found
-- Looking for daemon
-- Looking for daemon - found
-- Looking for 9 include files malloc.h, ..., unistd.h
-- Looking for 9 include files malloc.h, ..., unistd.h - found
-- Looking for 4 include files sys/socket.h, ..., sys/types.h
-- Looking for 4 include files sys/socket.h, ..., sys/types.h - found
-- Looking for 5 include files sys/types.h, ..., stdint.h
-- Looking for 5 include files sys/types.h, ..., stdint.h - found
-- Found BZip2: /usr/lib64/libbz2.so (found version "1.0.8") 
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11") 
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1i")  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for gettext
-- Looking for gettext - found
-- Found Iconv: /usr/lib64/libc.so  
-- Performing Test ICONV_SECOND_ARGUMENT_IS_CONST
-- Performing Test ICONV_SECOND_ARGUMENT_IS_CONST - Failed
-- Found Git: /usr/bin/git (found version "2.30.0") 
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found PCRE: /usr/lib64/libpcrecpp.so;/usr/lib64/libpcre.so  
-- Building with libattr support
CMake Warning (dev) at CMakeLists.txt:111 (find_package):
  Ignoring EXACT since no version is requested.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find Lua (missing: LUA_LIBRARIES LUA_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindLua.cmake:232 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:111 (find_package)


-- Configuring incomplete, errors occurred!
See also "/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build/CMakeFiles/CMakeOutput.log".
See also "/var/tmp/portage/net-p2p/eiskaltdcpp-2.4.1/work/eiskaltdcpp-2.4.1_build/CMakeFiles/CMakeError.log".
 * ERROR: net-p2p/eiskaltdcpp-2.4.1::testworld failed (configure phase):
 *   cmake failed
```

Reply:
This one will be a bit harder to fix, seems it still tries to detect lua even if lua is switched off.

Reply:
Found the issue, the old code didn't disable lua if the lua USE flag was disabled (because it now defaults to on).